x/expiration/types/params.go

```diff
@@ -71,27 +62,25 @@ func (p *Params) Equal(that interface{}) bool {
 			return false
 		}
 	}
-	if that1 == nil {
-		return p == nil
-	} else if p == nil {
-		return false
-	}
-	return true
+
+	return that1.Deposit.IsEqual(p.Deposit)
 }
 
+// Validate validates the deposit parameter
 func (p *Params) Validate() error {
 	return validateDepositParam(p.Deposit)
 }
 
+// Private method that runs validation on the deposit parameter
 func validateDepositParam(i interface{}) error {
 	coin, ok := i.(sdk.Coin)
 	if !ok {
 		return fmt.Errorf("invalid parameter type: %T", i)
 	}
 
 	// validate appropriate Coin
-	if coin.Validate() != nil {
-		return fmt.Errorf("invalid parameter type: %T", i)
+	if err := coin.Validate(); err != nil {
+		return fmt.Errorf("invalid parameter: %T", err)
```

Comment:
Need to use `%w` here instead of `%T` and maybe change it to `invalid coin` since we know it's a coin type.

Reply:
Updated it to `invalid coin`.

It doesn't like `%w`. It leads to a compilation error. `fmt.Errorf format %w has arg i of wrong type interface{}`

Reply:
`return fmt.Errorf("invalid parameter: %T", err)` does not contain the `i` variable.